### ISSUE TITLE
Stage 1 crashes when using std.log.warn in a function that returns a comptime_int

### ISSUE THREAD

**Issue by paoda** [ziglang/zig#10966]
### Zig Version

0.10.0-dev.876+628e9e6d0

### Steps to Reproduce

Here's a minimal example: 

```zig 
const std = @import("std");

pub fn main() anyerror!void {
    _ = this_crashes_the_compiler("Some Log Message", .{});
}

// Changing the Return type to a u32 (for example) prevents the bug from occuring.
fn this_crashes_the_compiler(comptime format: []const u8, args: anytype) comptime_int {
    std.log.warn(format, args);
    return 0;
}
```

### Expected Behavior

I expected Zig to either build or complain about this program.

### Actual Behavior

This is the Zig Compiler's Output: 
```
when analyzing /opt/zigup/zig/0.10.0-dev.876+628e9e6d0/files/lib/std/log.zig:133:38 in compiler source at /workspace/src/stage1/ir.cpp:5626: thread 26364 panic: assertion failed. This is a bug in the Zig compiler.
Unable to dump stack trace: debug info stripped
min-example...The following command terminated unexpectedly:
/opt/zigup/zig/0.10.0-dev.876+628e9e6d0/files/zig build-exe /home/paoda/Documents/dev/test/min-example/src/main.zig --cache-dir /home/paoda/Documents/dev/test/min-example/zig-cache --global-cache-dir /home/paoda/.cache/zig --name min-example --enable-cache 
error: the following build command failed with exit code 6:
/home/paoda/Documents/dev/test/min-example/zig-cache/o/fab784c0668ee1a0c570e698e2661a9c/build /opt/zigup/zig/0.10.0-dev.876+628e9e6d0/files/zig /home/paoda/Documents/dev/test/min-example /home/paoda/Documents/dev/test/min-example/zig-cache /home/paoda/.cache/zig
```

**Comment by paoda** [comment]
I believe this is a regression? `0.9.1` has this same issue but `0.8.0` outputs: 
```
/opt/zigup/zig/0.8.0/files/lib/std/Thread/Mutex.zig:99:17: error: unable to evaluate constant expression
        switch (@atomicRmw(State, &m.state, .Xchg, .locked, .Acquire)) {
                ^
/opt/zigup/zig/0.8.0/files/lib/std/Thread/Mutex.zig:50:26: note: called from here
    return m.impl.acquire();
                         ^
/opt/zigup/zig/0.8.0/files/lib/std/log.zig:146:60: note: called from here
            const held = std.debug.getStderrMutex().acquire();
                                                           ^
/opt/zigup/zig/0.8.0/files/lib/std/log.zig:205:16: note: called from here
            log(.warn, scope, format, args);
               ^
./main.zig:9:17: note: called from here
    std.log.warn(format, args);
                ^
./main.zig:4:34: note: called from here
    _ = this_crashes_the_compiler("Some Log Message", .{});
                                 ^
./main.zig:3:29: note: called from here
pub fn main() anyerror!void {
                            ^
/opt/zigup/zig/0.8.0/files/lib/std/Thread/Mutex.zig:50:26: note: referenced here
    return m.impl.acquire();
                         ^
/opt/zigup/zig/0.8.0/files/lib/std/log.zig:146:60: note: referenced here
            const held = std.debug.getStderrMutex().acquire();
                                                           ^
/opt/zigup/zig/0.8.0/files/lib/std/log.zig:205:16: note: referenced here
            log(.warn, scope, format, args);
               ^
./main.zig:9:17: note: referenced here
    std.log.warn(format, args);
                ^
./main.zig:4:34: note: referenced here
    _ = this_crashes_the_compiler("Some Log Message", .{});
                                 ^
min-example...The following command exited with error code 1:
/opt/zigup/zig/0.8.0/files/zig build-exe /home/paoda/Documents/dev/test/min-example/src/main.zig --cache-dir /home/paoda/Documents/dev/test/min-example/zig-cache --global-cache-dir /home/paoda/.cache/zig --name min-example --enable-cache 
error: the following build command failed with exit code 1:
/home/paoda/Documents/dev/test/min-example/zig-cache/o/35fcb16c686c4fa37d55fdaeb444e928/build /opt/zigup/zig/0.8.0/files/zig /home/paoda/Documents/dev/test/min-example /home/paoda/Documents/dev/test/min-example/zig-cache /home/paoda/.cache/zig
```

**Comment by paoda** [comment]
Was closed a while ago by ae8d26a6a0
